test/sse_test.dart

```diff
@@ -98,7 +114,7 @@ void main() {
       expect(handler.numberOfClients, 0);
     });
 
-    test('Client reconnects after being disconnected', () async {
+    test('client reconnects after being disconnected', () async {
```

Comment:
trying to send a few messages after a disconnect could repro the issue where I think an ack message to handle failed requests would be useful.

Reply:
A disconnect in this case only refers to the EventSource connection which is uni-directional. The client doesn't have a persistent connection to the server so there is no "disconnect". I do account for errors now so that we don't continually increment the id counter.